R/fit.R

```diff
@@ -599,7 +651,7 @@ CmdStanMCMC <- R6::R6Class(
         all = private$sampling_info_$model_params
       )
 
-      if (is.null(to_read) || (length(to_read) > 0)) {
+      if (is.null(to_read) || any(nzchar(to_read))) {
```

Comment:
Adding the test for the save_object method resulted in errors, and debugging led me to realize that  
`length(to_read) > 0` is TRUE if `to_read = ""`. For some reason in R `length("")` is 1 not 0 (I always forget that) so we need to use nzchar. 

This means that we were actually reading in everything again even if `to_read` was an empty string! Fixed now.

Reply:
Ah, didnt know that. I am learning my R as I go :)

Reply:
No worries!. I _should_ have known that but I always forget.